value.go

```diff
@@ -48,7 +48,7 @@ import (
 
 // maxVlogFileSize is the maximum size of the vlog file which can be created. Vlog Offset is of
 // uint32, so limiting at max uint32.
-var maxVlogFileSize = math.MaxUint32
+var maxVlogFileSize = int64(math.MaxUint32)
```

Comment:
```suggestion
var maxVlogFileSize uint32 = math.MaxUint32
```

Setting the variable type here would also fix the issue.

Reply:
@jarifibrahim This PR is part of fixing the 386 build issue, but the complete fix for that requires some additional changes.

I have a second PR that will provide that fix.  I would rather provide that separately, as in involves changing the amount of memory requested from mmap in multiple places, when running tests with `GOARCH=386`.  It demonstrates a fix to the problem, but may not be exactly in the form you would prefer.

Reply:
@gammazero that makes sense.
Do you think making the change I requested make sense?

Reply:
@jarifibrahim Committed suggestion - makes sense.